examples/permissioned-token/scripts/issuance/issue.js

```diff
@@ -0,0 +1,99 @@
+ * from that file, append his own signature & send it to network.
+ *  - Use `algob.executeSignedTxnFromFile` to execute tx from file
+ *  - In below function we assume creator & reserve is a single account (alice)
+ */
+async function issue (deployer, account, amount) {
+  const asaReserve = deployer.accountsByName.get('alice'); // asset reserve is the issuer
+
+  const asaInfo = deployer.asa.get('gold');
+  const controllerSSCInfo = deployer.getSSC('controller.py', 'clear_state_program.py');
+
+  const escrowParams = {
+    TOKEN_ID: asaInfo.assetIndex,
+    CONTROLLER_APP_ID: controllerSSCInfo.appID
+  };
+
+  const escrowLsig = await deployer.loadLogic('clawback.py', [], escrowParams);
+  const escrowAddress = escrowLsig.address();
+
+  const issuanceParams = [
+    /**
+     * tx 0 - Call to controller stateful smart contract with application arg: 'issue'
+     * The 'issue' branch ensures that sender is the token reserve and token is not killed
+     * Issuance tx will be rejected if token has been killed by the manager
+     */
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: asaReserve,
+      appId: controllerSSCInfo.appID,
+      payFlags: { totalFee: 1000 },
+      appArgs: ['str:issue'],
+      foreignAssets: [asaInfo.assetIndex]
+    },
+    /**
+     * tx 1 - Asset transfer transaction from Token Reserve -> account. This tx is executed
+     * and approved by the clawback escrow (since the asset is default frozen only clawback can move funds).
+     * This tx doesn't need rule checks (as this is an issuance txn)
+     */
+    {
+      type: types.TransactionType.RevokeAsset,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: escrowAddress,
+      recipient: account.addr,
+      assetID: asaInfo.assetIndex,
+      revocationTarget: asaReserve.addr, // tx will fail if assetSender is not token reserve address
+      amount: amount,
+      lsig: escrowLsig,
+      payFlags: { totalFee: 1000 }
+    }
+  ];
+  console.log(`* Issuing ${amount} tokens to ${account.name}:${account.addr} *`);
+  await executeTransaction(deployer, issuanceParams);
+
+  console.log(`* ${account.name} asset holding: *`);
+  await balanceOf(deployer, account.addr, asaInfo.assetIndex); // print asset holding
+
+  const supply = await totalSupply(deployer, asaInfo.assetIndex);
+  console.log(`Total Supply of token 'gold': ${supply}`);
+}
+
+async function run (runtimeEnv, deployer) {
+  const elon = deployer.accountsByName.get('elon-musk');
+  await fundAccount(deployer, elon); // fund elon
+
+  // opt-in asa to account first (skip if opted-in already)
+  console.log(`* Opt-In ASA gold for ${elon.name} *`);
+  await deployer.optInAcountToASA('gold', elon.name, {});
+  console.log('* Opt-In Successful *');
+
+  /**
+   * If using msig address as asa token reserve, then realistically user will receive a tx
+   * file signed by accounts <= threshold in a multisig group.
```

Comment:
```suggestion
   * file signed by accounts >= threshold in a multisig group.
```

Reply:
No. Current user also appends his signature, so it should be signed by <= threshold. As current user is also one of the accounts of multisig

Reply:
ah, right, I read it too fast. I thought it's to sign and send.